src/Connectivity/Connectivity.jl

```diff
@@ -293,6 +307,10 @@ function strongly_connected_components end
     return state.comps
 end
 
+struct Tarjan <: StrongConnectivityAlgorithm end
+
+@traitfn connected_components(g::::IsDirected, ::Tarjan) = strongly_connected_components(g)
```

Comment:
ok, but
1) we want to do the same thing for Kosaraju, and
2) just put the code for `strongly_connected_components` inside the Tarjan traitfn since we're not reusing it anywhere.

Reply:
some other algorithms use it, such transitive reduction and mean cycle

Reply:
We haven't migrated those yet so you'll need to fix the deprecations as well.